resources/wikia/modules/viewportObserver.js

```diff
@@ -0,0 +1,64 @@
+/*global define*/
+define('wikia.viewportObserver', [
+	'wikia.document',
+	'wikia.domCalculator',
+	'wikia.throttle',
+	'wikia.window'
+], function (doc, domCalculator, throttle, win) {
+	'use strict';
+
+	/**
+	 * Element is considered as in the viewport
+	 * when at least 50% of its height is in the viewport.
+	 * We take into account global navigation height as it covers page elements so they're not visible.
+	 */
+	function isInViewport(element) {
```

Comment:
I'm wondering about moving this method to `domCalculator` method. This way you don't have to make this method public in this module only for tests purposes. And this method is responsible for calculating the elements position.